packages/react-dom/src/server/ReactPartialRenderer.js

```diff
@@ -212,7 +212,8 @@ const processStyleName = function(styleName) {
   if (styleNameCache.hasOwnProperty(styleName)) {
     return styleNameCache[styleName];
   }
-  const result = hyphenateStyleName(styleName);
+  const isCustomProperty = styleName.indexOf('--') === 0;
```

Comment:
It's not ideal that we do this check again because `createMarkupForStyles` which calls us already does the same check anyway.